components/ca-rating.js

```diff
@@ -0,0 +1,202 @@
+         * @type {Event}
+         * @returns {undefined} nothing
+         */
+        attachedCallback() {
+            // render the component after it has been attached to the DOM
+            this.attached = true;
+            this.render();
+        }
+
+        /**
+         * @description Executes when any attribute is changed on the element
+         * @param {string} attrName - the name of the attribute to have changed
+         * @param {string} oldVal - the old value of the attribute
+         * @param {string} newVal - the new value of the attribute
+         * @returns {undefined}
+         */
+        attributeChangedCallback(attrName, oldVal, newVal) {
+            // only need to re-render the component if the attribute value has changed
+            if (oldVal !== newVal) {
+                this[attrName] = newVal;
+                this.render();
+            }
+        }
+
+        /** @returns {string} - gets the id of the element */
+        get id() {
+            return this.getAttribute('id');
+        }
+
+        /** @param {string} value - sets the id of the element */
+        set id(value) {
+            this.setAttribute('id', value);
+        }
+
+        /** @returns {string} - gets the name of the element */
+        get name() {
+            return this.getAttribute('name');
+        }
+
+        /** @param {string} value - sets the name of the element */
+        set name(value) {
+            this.setAttribute('name', value);
+        }
+
+        /** @returns {string} - gets the from value of the element */
+        get from() {
+            return parseInt(this.getAttribute('from'), 10) || this.defaults.from;
+        }
+
+        /** @param {string} value - sets the from value of the element */
+        set from(value) {
+            this.setAttribute('from', value || this.defaults.from);
+        }
+
+        /** @returns {string} to - gets the to value of the element */
+        get to() {
+            return parseInt(this.getAttribute('to'), 10) || this.defaults.to;
+        }
+
+        /** @param {string} value - sets the to value of the element */
+        set to(value) {
+            this.setAttribute('to', value || this.defaults.to);
+        }
+
+        /** @returns {string} value - gets the value of the element */
+        get value() {
+            return parseInt(this.getAttribute('value'), 10) || this.defaults.value;
+        }
+
+        /** @param {string} value - sets the value of the element */
+        set value(value) {
+            this.setAttribute('value', value || this.defaults.value);
+        }
+
+        /** @property {string} ca-rating.defaults - gets the defaults of the element */
+        get defaults() {
+            return {
+                from: 1,   // the lowest ranking value to be displayed
+                to: 10,    // the highest ranking value to be displayed
+                value: -1  // the initially selected ranking value
+            };
+        }
+
```

Comment:
The following attributes were not migrated:
attached: https://bitbucket.org/carepo/weblab-common-ui/src/b4002a5dd55bdc1e19a7b08cbf4bc6a7e12bb301/components/ca-rating.js?at=master&fileviewer=file-view-default#ca-rating.js-71

selectedItem: https://bitbucket.org/carepo/weblab-common-ui/src/b4002a5dd55bdc1e19a7b08cbf4bc6a7e12bb301/components/ca-rating.js?at=master&fileviewer=file-view-default#ca-rating.js-80